runtime/lua/vim/lsp/callbacks.lua

```diff
@@ -74,11 +74,10 @@ end
 
 --@see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_publishDiagnostics
 M['textDocument/publishDiagnostics'] = function(_, _, result)
-  if not result then return end
-  local uri = result.uri
-
   -- prevent creating/loading bufers for empty diagnostics
-  if vim.tbl_isempty(result.diagnostics) then return end
+  if not result or not result.diagnostics or vim.tbl_isempty(result.diagnostics) then return end
```

Comment:
Hmm  I don't know why need to check `not result.diagnostics`?

Reply:
maybe they send a `diagnostics` key with a `nil` value rather than an empty dict

Reply:
check the https://microsoft.github.io/language-server-protocol/specification#textDocument_publishDiagnostics.

Reply:
what does the actual servers do (they do not always follow the spec)

Reply:
I tried lua-languageserver and gopls, even without any diagnosis, `diagnostics` will be returned by the server with an empty table. I don't check other language servers.

Reply:
Same here, I tested CodeQL LS as well and it also return empty diagnostics dict. I think that checking if diagnostic dict exist in the first place is a safer approach though.